sdk/aqueduct/utils/local_data.py

```diff
@@ -0,0 +1,71 @@
+import os
+from typing import Optional
+
+from aqueduct.constants.enums import ArtifactType, LocalDataTableFormat
+from aqueduct.models.local_data import LocalData
+
+from ..error import InvalidUserArgumentException
+
+# The current local data maximum is temporarily set to 1MB.
+MAXIMUM_LOCAL_DATA_SIZE = 1000000
+
+
+def Local_Data(
+    path: str,
+    artifact_type: ArtifactType,
+    format: Optional[str] = None,
+) -> LocalData:
+    """Identify the local data which can be passed in as a parameter.
+
+    Args:
+        path:
+            The path to the data.
+        artifact_type:
+                The expected type of the data. The `ArtifactType` class in `enums.py` contains all
+                supported types, except for ArtifactType.UNTYPED.
+        format:
+                If the artifact type is ArtifactType.TABLE, the user has to specify the table format.
+                We currently support JSON, CSV, and Parquet.
+
+    Returns
+        A `LocalData` object which contains necessary information to create a parameter artifact.
+    """
+
+    format_enum = _convert_to_local_data_table_format(format)
+    return LocalData(path=path, artifact_type=artifact_type, format=format_enum)
+
+
+def validate_local_data(val: LocalData) -> None:
```

Comment:
This validation should be happening in the constructor of LocalData. The extraction of the data itself can happen at create_param() time.